pkg/proxy/util/nodeport_addresses.go

```diff
@@ -91,16 +90,17 @@ func (npa *NodePortAddresses) MatchAll() bool {
 	return npa.matchAll
 }
 
-// GetNodeAddresses return all matched node IP addresses for npa's CIDRs. If no matching
+// GetNodeIPs return all matched node IP addresses for npa's CIDRs. If no matching
```

Comment:
what is the benefit of returning net.IP? we have to convert them to string later to add it to the iptables rules

Reply:
Gotta check this again... it was definitely an optimization originally but possibly that got lost in refactoring...

Reply:
Given the current code, the only optimization is that it lets iptables/ipvs call `ip.IsLoopback()` rather than `utilproxy.IsLoopBack(address)`, saving an extra re-parsing. So not a huge improvement.

But in general, I feel like APIs involving IPs/CIDRs should use `net.IP`/`net.IPNet` rather than `string`, because if you have a string and you need the parsed form, then you have to either error-check the conversion or else leave readers worrying about the fact that you didn't (https://github.com/kubernetes/kubernetes/pull/116016#discussion_r1196143514). Whereas if you have a `net.IP`/`net.IPNet`, then `.String()` always works (and always guarantees you end up with the canonical form).

Reply:
I feel the same,